presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -518,6 +518,38 @@ public void commitCreateTable(ConnectorOutputTableHandle tableHandle, Collection
         metastore.createTable(table);
     }
 
+    @Override
+    public void rollbackCreateTable(ConnectorOutputTableHandle tableHandle)
+    {
+        HiveOutputTableHandle handle = checkType(tableHandle, HiveOutputTableHandle.class, "tableHandle");
+
+        // cleanup temporary path
+        if (handle.hasTemporaryPath()) {
+            Path temporaryPath = new Path(handle.getTemporaryPath());
+            if (pathExists(temporaryPath)) {
+                try {
+                    delete(temporaryPath, true);
+                }
+                catch (IOException e) {
+                    e.printStackTrace();
```

Comment:
I would not use printStackTrace, use logger instead.

Other than that I think that there should be no exception thrown here in normal situation, because you already checked if given path exist before. The only situation when we could expect this exception is when having multiple coordinators where multiple hiveMetadata is going to rollbackCreateTable concurrently, and one coordinator just removed this directory.

So instead of ```e.printStackTrace()``` I would write: 
```
catch(IOException e) {
    if (pathExist(temporaryPath) {
        throw new SomeProperPrestoMetadataException("Unable to delete temporary table data", e);
    }
    LOGGER.warn("During deletion of temporary table data", e);
}
```

What do you think about that?